plugins/modules/rds_cluster_param_group.py

```diff
@@ -0,0 +1,369 @@
+
+INT_MODIFIERS = {
+    "K": 1024,
+    "M": pow(1024, 2),
+    "G": pow(1024, 3),
+    "T": pow(1024, 4),
+}
+
+
+@AWSRetry.jittered_backoff()
+def _describe_db_cluster_parameters(module, connection, group_name):
+    try:
+        paginator = connection.get_paginator("describe_db_cluster_parameters")
+        return paginator.paginate(DBClusterParameterGroupName=group_name).build_full_result()
+    except is_boto3_error_code("DBParameterGroupNotFound"):
+        return None
+    except botocore.exceptions.ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to describe existing RDS cluster parameter groups")
+
+
+def convert_parameter(param, value):
+    """
+    Allows setting parameters with 10M = 10* 1024 * 1024 and so on.
+    """
+    converted_value = value
+
+    if param["DataType"] == "integer":
+        if isinstance(value, string_types):
+            try:
+                for modifier in INT_MODIFIERS.keys():
+                    if value.endswith(modifier):
+                        converted_value = int(value[:-1]) * INT_MODIFIERS[modifier]
+            except ValueError:
+                # may be based on a variable (ie. {foo*3/4}) so
+                # just pass it on through to the AWS SDK
+                pass
+        elif isinstance(value, bool):
+            converted_value = 1 if value else 0
+
+    elif param["DataType"] == "boolean":
+        if isinstance(value, string_types):
+            converted_value = value in BOOLEANS_TRUE
+        # convert True/False to 1/0
+        converted_value = 1 if converted_value else 0
+    return str(converted_value)
+
+
+def update_parameters(module, connection):
+    group_name = module.params["name"]
+    db_parameter_group_family = module.params["db_parameter_group_family"]
+    user_params = module.params["params"]
+    apply_method = "immediate" if module.params["immediate"] else "pending-reboot"
+    modify_list = []
+    db_cluster_params = []
+    response = _describe_db_cluster_parameters(module, connection, group_name)
+    if response:
+        db_cluster_params = response["Parameters"]
+    invalids, not_modifiable = [], []
+    for param_key, param_value in user_params.items():
+        found = list(filter(lambda x: x["ParameterName"] == param_key, db_cluster_params))
+        if not found:
+            invalids.append(param_key)
+            continue
+        param_def = found[0]
+        converted_value = convert_parameter(param_def, param_value)
+        # engine-default parameters do not have a ParameterValue, so we'll always override those.
+        if converted_value != param_def.get("ParameterValue"):
+            if param_def["IsModifiable"]:
+                modify_list.append(
+                    dict(ParameterValue=converted_value, ParameterName=param_key, ApplyMethod=apply_method)
+                )
+            else:
+                not_modifiable.append(param_key)
+    if not_modifiable or invalids:
```

Comment:
Can we handle these conditions with the following exceptions as given in https://boto3.amazonaws.com/v1/documentation/api/1.26.93/reference/services/rds/client/modify_db_parameter_group.html?

RDS.Client.exceptions.DBParameterGroupNotFoundFault
RDS.Client.exceptions.InvalidDBParameterGroupStateFault

Reply:
updated